backend-api/src/functions/adapters/kmsAdapter.ts

```diff
@@ -3,53 +3,41 @@ import {
   DecryptCommandOutput,
   KMSClient,
 } from "@aws-sdk/client-kms";
-import { errorResult, Result, successResult } from "../utils/result";
 import { NodeHttpHandler } from "@smithy/node-http-handler";
 
-export class KMSAdapter implements IKmsAdapter {
-  private readonly kmsClient = new KMSClient({
-    region: "eu-west-2",
-    maxAttempts: 2,
-    requestHandler: new NodeHttpHandler({
-      connectionTimeout: 5000,
-      requestTimeout: 5000,
-    }),
-  });
+const kmsClient = new KMSClient({
+  region: "eu-west-2",
+  maxAttempts: 2,
+  requestHandler: new NodeHttpHandler({
+    connectionTimeout: 5000,
+    requestTimeout: 5000,
+  }),
+});
 
+export class KMSAdapter implements IDecryptAsymmetric {
   async decrypt(
```

Comment:
Throw errors instead of trying to handle them here. The errors will be caught and handled by the calling method.